File: Python/underautomation/universal_robots/xml_rpc/internal/xml_rpc_server_base.py
import typing
from underautomation.universal_robots.internal.ur_service_base import URServiceBase
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.XmlRpc.Internal import XmlRpcServerBase as xml_rpc_server_base

class XmlRpcServerBase(URServiceBase):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = xml_rpc_server_base()
		else:
			self._instance = _internal
	def xml_rpc_server_request(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.XmlRpcServerRequest+= lambda sender, request : handler(Wrapper(sender), Wrapper(request))
	def start(self, port: int) -> None:
		self._instance.Start(port)
	def stop(self) -> None:
		self._instance.Stop()
	@property
	def enabled(self) -> bool:
		return self._instance.Enabled
	@property
	def port(self) -> int:
		return self._instance.Port

File: UnderAutomation.UniversalRobots.ObfuscatedSources/XmlRpc/Internal/XmlRpcServerBase.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Internal;
using XmlRpc;

namespace XmlRpc.Internal {

	public abstract class XmlRpcServerBase : URServiceBase {

		/// <summary>
		/// Event raised when a XML-RPC request is sent by the robot and received.
		/// </summary>
		/// <param name="sender">The UR instance that raised this event</param>
		/// <param name="request">Content of the request</param>
		public delegate void XmlRpcServerRequestEventHandler(object sender, XmlRpcEventArg request);

		/// <summary>
		/// Enable the local XML-RPC server to receive commands from the robot
		/// </summary>
		/// <param name="port">Port number on which the server should start</param>
		public void Start(int port)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Disable and close the socket used for the XML-RPC server
		/// </summary>
		public void Stop()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Is the XML-RPC server enabled
		/// </summary>
		public bool Enabled { get; }

		/// <summary>
		/// Local port on which the XML-RPC server is running. 0 if server is disabled
		/// </summary>
		public int Port { get; }

		/// <summary>
		/// Event raised when a XML-RPC request has been sent from the robot to this machine. You should answer to the robot in this event via the property request.Answer
		/// </summary>
		public event XmlRpcServerBase.XmlRpcServerRequestEventHandler XmlRpcServerRequest;
	}
}
